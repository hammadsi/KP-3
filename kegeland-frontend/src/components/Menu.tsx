import { VStack, Text, Divider, HStack, Icon, Box } from '@chakra-ui/react';
import { ComponentProps } from 'react';
import { IconType } from 'react-icons/lib';
import { Link, To } from 'react-router-dom';

import { ThemeMode } from '../types';

type MenuItemProps = {
  title: string;
  mode?: ThemeMode;
  to: To;
  icon?: IconType;
};

const Item: React.FC<MenuItemProps> = ({ title, mode, to, icon }) => {
  const isDark = mode === 'dark';
  return (
    <HStack
      spacing={1}
      marginBottom={2}
      width="full"
      _hover={{ cursor: 'pointer' }}>
      {icon && <Icon as={icon} color="primary.600" fontSize={24} />}
      <Link to={to}>
        <Text
          fontWeight="semibold"
          transition="all 300ms"
          color={isDark ? 'gray.200' : 'gray.700'}
          _hover={{ color: isDark ? 'gray.400' : 'gray.500' }}>
          {title}
        </Text>
      </Link>
    </HStack>
  );
};

Item.defaultProps = {
  mode: 'dark',
};

type MenuComponents = {
  Item: typeof Item;
};

type MenuProps = {
  title: string;
  mode?: ThemeMode;
  head?: React.ReactNode;
  children?: React.ReactNode;
} & ComponentProps<typeof VStack>;

const Menu: React.FunctionComponent<MenuProps> & MenuComponents = ({
  title,
  mode,
  head,
  children,
  ...props
}) => {
  const isDark = mode === 'dark';
  return (
    <VStack
      {...props}
      width="full"
      spacing={2}
      paddingTop={4}
      alignItems="flex-start">
      {head && <Box paddingBottom={2}>{head}</Box>}
      {head && (
        <Divider borderColor={isDark ? 'whiteAlpha.400' : 'blackAlpha.400'} />
      )}
      <Text
        color="primary.600"
        fontWeight="semibold"
        width="full"
        textTransform="uppercase">
        {title}
      </Text>
      {children}
      <Divider borderColor={isDark ? 'whiteAlpha.400' : 'blackAlpha.400'} />
    </VStack>
  );
};

Menu.defaultProps = {
  mode: 'dark',
};

Menu.Item = Item;

export default Menu;
