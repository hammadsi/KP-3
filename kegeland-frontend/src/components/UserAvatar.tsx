import { Wrap, WrapItem, Avatar, Text } from '@chakra-ui/react';
import { AiOutlineUser } from 'react-icons/ai';

import { User } from '../state/ducks/auth/auth.interface';
import { ThemeMode } from '../types';
import { renderName } from '../utils/renderName';

type UserAvatarProps = {
  user: User;
  mode?: ThemeMode;
};

const UserAvatar: React.FC<UserAvatarProps> = ({ user, mode }) => {
  const isDark = mode === 'dark';
  return (
    <Wrap>
      <WrapItem alignItems="center">
        <Avatar
          size="xs"
          bg="primary.600"
          color="white"
          name={renderName(user.name)}
          icon={<AiOutlineUser />}
        />
        <Text
          paddingLeft={1}
          fontWeight={'semibold'}
          color={isDark ? 'gray.200' : 'gray.700'}
        >
          {renderName(user.name)}
        </Text>
      </WrapItem>
    </Wrap>
  );
};

UserAvatar.defaultProps = {
  mode: 'dark',
};

export default UserAvatar;
