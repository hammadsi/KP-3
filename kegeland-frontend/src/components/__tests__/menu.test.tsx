import renderer from 'react-test-renderer';

import Menu from '../Menu';
import { userMock } from '../mocks/user.mock';
import UserAvatar from '../UserAvatar';

describe('Test manu', () => {
  it('renders correctly dark', () => {
    const tree = renderer.create(<Menu title={'hello'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly ligth', () => {
    const tree = renderer
      .create(
        <Menu
          title={'hello'}
          mode={'light'}
          head={<UserAvatar user={userMock} />}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
