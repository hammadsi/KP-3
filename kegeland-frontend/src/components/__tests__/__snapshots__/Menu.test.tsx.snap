// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test manu renders correctly dark 1`] = `
<div
  className="chakra-stack css-7nrznn"
>
  <p
    className="chakra-text css-199kdpe"
  >
    hello
  </p>
  <hr
    aria-orientation="horizontal"
    className="chakra-divider css-1x4iw6"
  />
</div>
`;

exports[`Test manu renders correctly ligth 1`] = `
<div
  className="chakra-stack css-7nrznn"
>
  <div
    className="css-1drbkg9"
  >
    <div
      className="chakra-wrap css-i6bazn"
    >
      <ul
        className="chakra-wrap__list css-wah4g8"
      >
        <li
          className="chakra-wrap__listitem css-70qvj9"
        >
          <span
            className="chakra-avatar css-ew7jl2"
          >
            <div
              aria-label="ola Nordmann"
              className="chakra-avatar__initials css-0"
              role="img"
            >
              oN
            </div>
          </span>
          <p
            className="chakra-text css-zyu9tc"
          >
            ola Nordmann
          </p>
        </li>
      </ul>
    </div>
  </div>
  <hr
    aria-orientation="horizontal"
    className="chakra-divider css-nbrn3x"
  />
  <p
    className="chakra-text css-199kdpe"
  >
    hello
  </p>
  <hr
    aria-orientation="horizontal"
    className="chakra-divider css-nbrn3x"
  />
</div>
`;
